app/scripts/controllers/permissions/index.js

```diff
@@ -423,6 +432,40 @@ export class PermissionsController {
     })
   }
 
+  /**
+   * When a new account is selected in the UI, emit accountsChanged to each origin
+   * where the account order has changed.
+   *
+   * Note: This will emit "false positive" accountsChanged events, but they are
+   * handled by the inpage provider.
```

Comment:
nit: I think it would be more correct to say that `_handleConnectedAccountSelected` below does this.

Reply:
Yeah, I wasn't sure where it should go so I left it in both places 🤷‍♂️ This seems fine for now.